R/thumbnail.R

```diff
@@ -0,0 +1,233 @@
+#' Get the Content Thumbnail
+#'
+#' `get_thumbnail` saves the content thumbnail to the given path (default: temp file).
+#' `delete_thumbnail` removes the thumbnail (optionally saving to the given path)
+#' `has_thumbnail` returns whether the content has an thumbnail
+#'
+#' @param content A content object
+#' @param path optional. The path to the image on disk
+#'
+#' @rdname get_thumbnail
+#' @family content functions
+#' @export
+get_thumbnail <- function(content, path = NULL) {
+  validate_R6_class(content, "Content")
+  guid <- content$get_content()$guid
+
+  con <- content$get_connect()
+
+  # Connect 2024.09.0 introduced public endpoints for content thumbnails. We
+  # prefer those, falling back to the unversioned endpoints if unavailable.
+  res <- con$GET(
+    url = con$server_url("content", guid, "__thumbnail__"),
+    parser = NULL
+  )
+  if (httr::status_code(res) == 404) {
+    res <- con$GET(
+      unversioned_url("applications", guid, "image"),
+      parser = NULL
+    )
+  }
+
+  if (httr::status_code(res) == 204) {
+    return(NA)
+  }
+
+  # guess file extension
+  if (is.null(path)) {
+    ct <- httr::headers(res)$`content-type`
+    if (grepl("^image/", ct)) {
+      # just strip off 'image/'
+      ext <- substr(ct, 7, nchar(ct))
+      path <- fs::file_temp(pattern = "content_image_", ext = ext)
+    } else {
+      # try png
+      warning(glue::glue("Could not infer file extension from content type: {ct}. Using '.png'"))
+      path <- fs::file_temp(pattern = "content_image_", ext = ".png")
+    }
+  }
+
+  writeBin(httr::content(res, as = "raw"), path)
+
+  return(fs::as_fs_path(path))
```

Comment:
I'd leave it to the caller to `as_fs_path()` if they want

```suggestion
  path
```